sklearn/tree/export.py

```diff
@@ -415,3 +416,108 @@ def recurse(tree, node_id, criterion, parent=None, depth=0):
+         node_ids=False, proportion=False, rotate=False,
+         rounded=False, special_characters=False, out='matplotlib'):
+    """Plots decision tree.
+
+    Parameters
+    ----------
+    decision_tree : decision tree classifier
+        The decision tree to be plotted.
+
+    width : width of resulting tree.
+
+    height : height of resulting tree.
+
+    max_depth : int, optional (default=None)
+        The maximum depth of the representation. If None, the tree is fully
+        generated.
+
+    feature_names : list of strings, optional (default=None)
+        Names of each of the features.
+
+    class_names : list of strings, bool or None, optional (default=None)
+        Names of each of the target classes in ascending numerical order.
+        Only relevant for classification and not supported for multi-output.
+        If ``True``, shows a symbolic representation of the class name.
+
+    label : {'all', 'root', 'none'}, optional (default='all')
+        Whether to show informative labels for impurity, etc.
+        Options include 'all' to show at every node, 'root' to show only at
+        the top root node, or 'none' to not show at any node.
+
+    filled : bool, optional (default=False)
+        When set to ``True``, paint nodes to indicate majority class for
+        classification, extremity of values for regression, or purity of node
+        for multi-output.
+
+    leaves_parallel : bool, optional (default=False)
+        When set to ``True``, draw all leaf nodes at the bottom of the tree.
+
+    impurity : bool, optional (default=True)
+        When set to ``True``, show the impurity at each node.
+
+    node_ids : bool, optional (default=False)
+        When set to ``True``, show the ID number on each node.
+
+    proportion : bool, optional (default=False)
+        When set to ``True``, change the display of 'values' and/or 'samples'
+        to be proportions and percentages respectively.
+
+    rotate : bool, optional (default=False)
+        When set to ``True``, orient tree left to right rather than top-down.
+
+    rounded : bool, optional (default=False)
+        When set to ``True``, draw node boxes with rounded corners and use
+        Helvetica fonts instead of Times-Roman.
+
+    special_characters : bool, optional (default=False)
+        When set to ``False``, ignore special characters for PostScript
+        compatibility.
+
+    out : {'matplotlib', 'jupyter'}, optional (default='matplotlib')
+        Options include 'matplotlib' to plot using matplotlib library,
+        'jupyter' to plot to Jupyter Notebook environment.
+    """
+    if out not in ('matplotlib', 'jupyter'):
+        raise ValueError(
+            "Incorrect 'out' value passed (%s). "
+            "Use 'jupyter' or 'matplotlib'." % out)
+    try:
+        import graphviz
+    except ImportError:
+        raise ImportError('Plotting requires graphviz to be installed.')
+
+    in_memory_dot_file = six.StringIO()
+    export_graphviz(
```

Comment:
Where `None` returns and `<file_path>` writes to file

and 'image' returns/writes to file the image object returned by `graphviz.Source(...).pipe(format='')` and 'dot' returns/writes to file the dot formatted graph string.